src/core/layertreemodel.cpp

```diff
@@ -440,7 +440,7 @@ QVariant FlatLayerTreeModelBase::data( const QModelIndex &index, int role ) cons
     return QVariant();
 
   const QModelIndex sourceIndex = mapToSource( index );
-  if ( !sourceIndex.isValid() )
+  if ( !sourceIndex.isValid() || !sourceIndex.internalPointer() )
```

Comment:
The crash is linked to a casting of internalPointer():
![image](https://github.com/opengisch/QField/assets/1728657/6f41619c-7be0-4d57-8416-013f23829a69)

Which happens in index2node:
```
QgsLayerTreeNode *QgsLayerTreeModel::index2node( const QModelIndex &index ) const
{
  if ( !index.isValid() )
    return mRootNode;

  QObject *obj = reinterpret_cast<QObject *>( index.internalPointer() );
  return qobject_cast<QgsLayerTreeNode *>( obj );
}
```

Maybe checking/guarding from a null pointer will help.